simplecpp.cpp

```diff
@@ -2719,14 +2719,38 @@ static std::string toAbsolutePath(const std::string& path) {
     return simplecpp::simplifyPath(path);
 }
 
-static std::pair<std::string, bool> extractRelativePathFromAbsolute(const std::string& absolutepath) {
-    static const std::string prefix = currentDirectory() + "/";
-    if (startsWith_(absolutepath, prefix)) {
-        const std::size_t size = prefix.size();
-        return std::make_pair(absolutepath.substr(size, absolutepath.size() - size), true);
+// templated function for compiler optimization opportunities
```

Comment:
I hesitate to use templates unless the reason is to make code generic and avoid copy paste.

Reply:
OK, there will be no need for that on the final version. I'm working on fixing some crucial errors, and have more detailed tests, so see the next version.